src/hooks/tradingView/useTradingViewLimitOrder.ts

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+import BigNumber from 'bignumber.js';
+import { ContextMenuItem } from 'public/tradingview/charting_library';
+
+import { AbacusOrderSide, TradeInputField } from '@/constants/abacus';
+import { AnalyticsEvents } from '@/constants/analytics';
+import { STRING_KEYS } from '@/constants/localization';
+import { USD_DECIMALS } from '@/constants/numbers';
+import { StatsigFlags } from '@/constants/statsig';
+import { TradeTypes } from '@/constants/trade';
+
+import { getIsAccountConnected } from '@/state/accountSelectors';
+import { useAppDispatch, useAppSelector } from '@/state/appTypes';
+import { setTradeFormInputs } from '@/state/inputs';
+
+import abacusStateManager from '@/lib/abacus';
+import { track } from '@/lib/analytics/analytics';
+
+import { useStatsigGateValue } from '../useStatsig';
+import { useStringGetter } from '../useStringGetter';
+
+export function useTradingViewLimitOrder(
+  marketId?: string
+): (unixTime: number, price: number) => ContextMenuItem[] {
+  const dispatch = useAppDispatch();
+  const stringGetter = useStringGetter();
+  const canDraftLimitOrders = useStatsigGateValue(StatsigFlags.ffLimitOrdersFromChart);
+
+  // Every time we call tvChartWidget.onContextMenu, a new callback is _added_ and there is no way to remove previously
+  // added menu options. So, instead of creating a new callback and calling .onContextMenu every time `isUserConnected` changes,
+  // only pass in one stable callback on chart load that refers to changing `isUserConnected` values through a ref
+  const isUserConnected = useAppSelector(getIsAccountConnected);
```

Comment:
do we actually need to care about whether is user connected? i think we can still let logged out users to draft an order